src/libraries/System.Private.CoreLib/src/System/Globalization/CharUnicodeInfo.cs

```diff
@@ -273,13 +273,13 @@ internal static uint ToUpper(uint codePoint)
 
             nuint offset = GetCategoryCasingTableOffsetNoBoundsChecks(codePoint);
 
-            // The offset is specified in shorts:
-            // Get the 'ref short' corresponding to where the addend is, read it as a signed 16-bit value, then add
+            // The offset is specified in ushort:
+            // Get the 'ref ushort' corresponding to where the addend is, read it as a unsigned 16-bit value, then add
```

Comment:
This needs a comment along the lines that we're using 16-bit values because we need to stay within the same Unicode plane. Otherwise it's not obvious as to why the masking step on the final line exists.